internal/storage/localdir.go

```diff
@@ -25,15 +28,36 @@ const DefaultBundleCacheDir = "/var/cache/bundles"
 type LocalDirectory struct {
 	RootDirectory string
 	URL           url.URL
+	BearerToken   string
 }
 
-func (s *LocalDirectory) Load(_ context.Context, owner client.Object) (fs.FS, error) {
-	bundleFile, err := os.Open(s.bundlePath(owner.GetName()))
+func (s *LocalDirectory) Load(ctx context.Context, owner client.Object) (fs.FS, error) {
+	bundle := owner.(*rukpakv1alpha1.Bundle)
+
+	// FIXME(TODO): use a split storage? i.e. if the bundle is a plain bundle,
+	//  we know its "ours" and have it stored locally, so we can just load from
+	//  disk. If it isn't ours, we need to pull from its content URL. For now,
+	//  always pulling from the content URL should be fine.
```

Comment:
How bad of an idea would it be to first check local storage by stating the expected tar.gz file, and then fall back to network requests when we get a cache miss?

Reply:
Solved via #401